compiler/noirc_evaluator/src/ssa/ir/dfg/simplify/binary.rs

```diff
@@ -211,6 +219,22 @@ pub(super) fn simplify_binary(binary: &Binary, dfg: &mut DataFlowGraph) -> Simpl
                         lhs,
                         zero,
                     ));
+                } else if lhs_is_zero && dfg.runtime.is_acir() {
+                    // `0 < rhs` for unsigned values is the same as `rhs != 0`,
+                    // which is slightly more performant in ACIR
+                    let zero = dfg.make_constant(FieldElement::zero(), lhs_type);
+                    let instruction =
+                        Instruction::Binary(Binary { lhs: rhs, rhs: zero, operator: BinaryOp::Eq });
+                    let eq = dfg.insert_instruction_and_results(
+                        instruction,
+                        block,
+                        ctrl_typevars,
```

Comment:
```suggestion
                        Some(Type::bool()),
```
Whether ctrl_typevars are required is determined here https://github.com/noir-lang/noir/blob/8d787871c097c315193fed23204e56fa396003a7/compiler/noirc_evaluator/src/ssa/ir/instruction.rs#L446

For a binary we can see the following result_type: https://github.com/noir-lang/noir/blob/8d787871c097c315193fed23204e56fa396003a7/compiler/noirc_evaluator/src/ssa/ir/instruction/binary.rs#L82

Taking the `ctrl_typevars` from the original instruction would give us the incorrect type vars for `Eq`.

Reply:
I changed it to `None` as it seems binary instructions don't require them...?

Reply:
Based off this: https://github.com/noir-lang/noir/blob/8d787871c097c315193fed23204e56fa396003a7/compiler/noirc_evaluator/src/ssa/ir/instruction.rs#L446) 

It looks like they would require them? They just don't require passing along the `ctrl_typevars` because we know them.

Reply:
Oh yup, looks like it does!